### DISCUSSION TITLE
I am looking for someone to help implement a solver

### DISCUSSION THREAD

**Discussion by danielflam** [TimefoldAI/timefold-solver#1049] [General]
I have the basic structure of what i want but due to lack of time I need someone to come in and finish the implementation. obviously its a paid gig - can anyone point me in the right direction? this is a specialized field and i dont think ill find someone random on the internet that will be able to do this efficiently?

**Comment by kentbill** [comment]
hello, Daniel,
I am a freelancer who has done some planning-related projects using optaplanner/timefold, mainly in the business areas of production planning, VRP, etc. Currently, I mainly do planning-related projects. You can contact me via email (kentbill@gmail.com) and provide your project details to determine whether I can take over your project.
thank you!

**Comment by mcimbora** [comment]
Hi Daniel,

Thanks for reaching out. Timefold offers such services directly / via our partner network.

Feel free to contact us at https://timefold.ai/contact and we'll be in touch with you.

Regards,
Matej

**Comment by danielflam** [comment]
wow i missed this email.

my project is pretty small.
i built a  solver using ai
that from what i am seeing in the code
describes my problem pretty well.
however its basically scaffolding
so even after I managed to fix it and compile it, but it crashes on run,
and the i/o and ui is lacking.

what i want to do is
1) get the current code (or something similar) to run and solve *something*
2) improve the constraints so i have finer control over them
3) create an ability to enter the data table via a csv import or even
better directly from google sheets
4) nice to have is a web ui similar to their demos.

here is the description i entered in claude.ai (so the definition  of the
optimization)

"I want a java timefold  application that has  the following requirements
> There are students and teachers.
> We are paring  teachers and students in timeslots. Each timeslot is a
> group of up to 6 students (hard constraint).
> - Each student has a name, age, and a level. - There is a hard constraint
> of student available time slots.
> - There are are soft constraint preference of which other students they
> wants to be with.
> -  In addition students younger than 18 should be paired
> as close to their age, and level (soft constraint).
> - Students above 18 should not be paired with students under 18.
> - Each teacher has his availability as a hard constraint, and preferred
> time slots as a soft constraint.
> The application should use a ConstraintProvider and justifications
> There should be a demo data generator
> The program should print out the result in an easy to read format
> the solution class is called "TimeTable"


I then used the code and merged it into the hello world and changed it
based on the documentation
So I think this is a pretty good description of the problem.

Here is where i think this solution is currently lacking:
1) I would like control over the number of groups - so i wou say want to
eventually add hard constraints over the numer of
groups  - so say I want to have a hard limit on say 2 children groups of
level 3.
